java/src/main/java/com/avatao/tfw/tfwconnector/Serializer.java

```diff
@@ -0,0 +1,85 @@
+/* 
+ * Copyright (C) 2018 Avatao.com Innovative Learning Kft.
+ * All Rights Reserved. See LICENSE file for details.
+ */
+
+package com.avatao.tfw.tfwconnector;
+
+import java.io.UnsupportedEncodingException;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+/*
+ * TFW JSON message format
+ * 
+ * message:
+ * {
+ *     "key":     string,    # addressing
+ *     "data":    {...},     # payload
+ *     "trigger": string     # FSM trigger
+ * }
+ * 
+ * ZeroMQ's sub-pub sockets use enveloped messages
+ * (http://zguide.zeromq.org/page:all#Pub-Sub-Message-Envelopes)
+ * and TFW also uses them internally. This means that on ZMQ sockets
+ * we always send the messages key separately and then the actual
+ * message (which contains the key as well) like so:
+ * socket.send_multipart([message['key'], message])
+ * The purpose of this module is abstracting away this low level behaviour.
+ */
+
+/**
+ * A serializer class which transforms JSON objects to 
+ * TFW message format and vice versa.
+ */
+public class Serializer {
+
+    /**
+     * Serializes a JSON message into a multipart TFW message.
+     * @param message message to be serialized for TFW server
+     * @return multipart TFW message
+     */
+    public static List<String> serializeTFWMessage(ObjectNode message) {
```

Comment:
This method is basically pointless (the implementation of `serialize` should just simply go here)

Reply:
I merged serialize() into it.